### PR TITLE
MGMT-10780: Add CAPI+Hive jobs for OCM-2.5 branch

### PR DESCRIPTION
**Pull request by mkowalski**
/cc @eranco74

### CODE REVIEW

**Comment by mkowalski** [comment]
Waiting for https://github.com/openshift/assisted-service/pull/3940

**Comment by mkowalski** [comment]
```
 + oc get nodepool/assisted-test-cluster --selector= --namespace=assisted-spoke-cluster
+ grep -ivE '(no resources found|not found)'
NAME                    CLUSTER                 DESIRED NODES   CURRENT NODES   AUTOSCALING   AUTOREPAIR   VERSION       UPDATINGVERSION   UPDATINGCONFIG   MESSAGE
assisted-test-cluster   assisted-test-cluster   0                               False         False        4.11.0-fc.0                                      
```

This means we need https://github.com/openshift/assisted-service/pull/3937 and its cherrypick for this test to be reasonable

**Comment by mkowalski** [comment]
In the 2nd rehearse we have a success

```
+ oc get nodepool/assisted-test-cluster --selector= --namespace=assisted-spoke-cluster
+ grep -ivE '(no resources found|not found)'
NAME                    CLUSTER                 DESIRED NODES   CURRENT NODES   AUTOSCALING   AUTOREPAIR   VERSION       UPDATINGVERSION   UPDATINGCONFIG   MESSAGE
assisted-test-cluster   assisted-test-cluster   0                               False         False        4.11.0-fc.0           
                           
[...]

+ oc get hostedcluster/assisted-test-cluster --selector= --namespace=assisted-spoke-cluster
+ grep -ivE '(no resources found|not found)'
NAME                    VERSION   KUBECONFIG                               PROGRESS   AVAILABLE   REASON                    MESSAGE
assisted-test-cluster             assisted-test-cluster-admin-kubeconfig   Partial    True        HostedClusterAsExpected   

[...]

+ echo 'Scaling the hosted cluster up to contain 1 worker nodes'
+ oc scale nodepool/assisted-test-cluster -n assisted-spoke-cluster --replicas=1
nodepool.hypershift.openshift.io/assisted-test-cluster scaled

[...]

Worker nodes have been detected successfuly in the created cluster!
```

@osherdp, can we please merge that? The test itself is well defined, the intermittent failures come from the underlying system under test and will be fixed in the respective repos

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5-periodics.yaml
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5-periodics.yaml:261]:
```suggestion
  reporter_config:
    slack:
      channel: '#assisted-deployment-ci'
      job_states_to_report:
      - failure
      - error
      report_template: '{{ if eq .Status.State "success" }} :green_jenkins_circle:
        {{ else }} :red_jenkins_circle: {{ end }} Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
        <{{.Status.URL}}|View logs>'
  spec:
```

**Review by osherdp — CHANGES_REQUESTED** [review]
sure thing, just one small thing

**Comment by osherdp** [comment]
Sorry, one more thing
Seeing all those flakes in capi workflow, we disabled periodic jobs until we reach stability: https://github.com/openshift/release/pull/29508

Can you please change that too to match the 2.5 branch?

**Comment by mkowalski** [comment]
/hold

Let's wait for https://coreos.slack.com/archives/C014N2VLTQE/p1655104912244059 to understand whether the issue in the other set of jobs was hypershift or test-infra. If hypershift, then it's valid to disable periodics here too. If test-infra, it should not have an impact

**Comment by osherdp** [comment]
> /hold
> 
> Let's wait for https://coreos.slack.com/archives/C014N2VLTQE/p1655104912244059 to understand whether the issue in the other set of jobs was hypershift or test-infra. If hypershift, then it's valid to disable periodics here too. If test-infra, it should not have an impact

Sounds good

code itself is good so
/lgtm
/approve

**Comment by mkowalski** [comment]
Let's implement https://issues.redhat.com/browse/MGMT-10864 first so that for failures we see a bit more info

**Comment by mkowalski** [comment]
/retest

We have CAPI logs collector now implemented so let's check

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5-periodics.yaml
Comment (adriengentil) [inline ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5-periodics.yaml:260]:
```suggestion
  name: periodic-ci-openshift-assisted-service-release-ocm-2.5-e2e-metal-assisted-operator-capi-periodic
  reporter_config:
    slack:
      channel: '#assisted-deployment-ci'
      job_states_to_report:
      - failure
      - error
      report_template: '{{ if eq .Status.State "success" }} :green_jenkins_circle:
        {{ else }} :red_jenkins_circle: {{ end }} Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
        <{{.Status.URL}}|View logs>'
```
I think we need to add this block back

**Review by adriengentil — CHANGES_REQUESTED** [review]

**Comment by mkowalski** [comment]
Once this is merged, we need https://github.com/openshift/assisted-service/pull/3965

**Pull request merged**
